pkg/virt-launcher/virtwrap/api/schema.go

```diff
@@ -573,16 +574,37 @@ type Ballooning struct {
 	Model string `xml:"model,attr"`
 }
 
-type RandomGenerator struct {
-}
-
 type Watchdog struct {
 	Model  string `xml:"model,attr"`
 	Action string `xml:"action,attr"`
 	Alias  *Alias `xml:"alias,omitempty"`
 }
 
-// TODO ballooning, rng, cpu ...
+// Rng represents the source of entropy from host to VM
+type Rng struct {
+	// Model attribute specifies what type of RNG device is provided
+	Model string `xml:"model,attr"`
+	// Backend specifies the source of entropy to be used
+	Backend *RngBackend `xml:"backend,omitempty"`
+}
+
+// RngRate sets the limiting factor how to read from entropy source
+type RngRate struct {
```

Comment:
I think you do not specify it, libvirt will set defaults one, so at least for now we do not need to introduce it.